runtime/protocol/http-client/jvm/test/aws/smithy/kotlin/runtime/http/interceptors/requestcompression/GzipCompressionTest.kt

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.interceptors.requestcompression
+
+import aws.smithy.kotlin.runtime.hashing.crc32
+import aws.smithy.kotlin.runtime.http.interceptors.decompressGzipBytes
+import aws.smithy.kotlin.runtime.io.*
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+
+class GzipCompressionTest {
+    @Test
+    fun testBytesCompression() {
+        runBytesCompressionTest("")
+        runBytesCompressionTest("<Foo>bar</Foo>")
+        runBytesCompressionTest("<Baz>foo</Baz>".repeat(100))
+    }
+
+    @Test
+    fun testGzipSdkSourceCompression() = runTest {
+        // "readFully" implementation doesn't allow for a read() on empty payloads and will fail test because of that so skipping it
+        runGzipSdkSourceCompressionTest("", 1L, true)
+        runGzipSdkSourceCompressionTest("", 1001L, true)
+
+        runGzipSdkSourceCompressionTest("<Qux>baz</Qux>", 1L)
+        runGzipSdkSourceCompressionTest("<Qux>baz</Qux>", 2L)
+        runGzipSdkSourceCompressionTest("<Bar>foo</Bar>", 1002L)
+
+        runGzipSdkSourceCompressionTest(
+            "“Foo Bar” are very commonly seen as variables in samples and examples. Some sources will claim " +
+                "that this is an interpretation of “F.U.B.A.R.”, a military term...",
+            3L,
+        )
+        runGzipSdkSourceCompressionTest(
+            "“Foo Bar” are very commonly seen as variables in samples and examples. Some sources will claim " +
+                "that this is an interpretation of “F.U.B.A.R.”, a military term...",
+            1003L,
+        )
+
+        runGzipSdkSourceCompressionTest(
+            "Lorem ipsum dolor sit amet, dolorem corpora iracundia has ea, duo cu stet alterum scriptorem, " +
+                "et qui putent tractatos. Ne epicurei gloriatur pro, et ornatus consulatu necessitatibus qui. " +
+                "Veri eripuit feugiat sed no, dicat ridens id quo. Mei ne putent impedit antiopam. Ad libris " +
+                "assueverit his. Quo te brute vitae iuvaret, ut nibh bonorum sea. Mel altera vocibus ei, no vel" +
+                " tantas postea doming.",
+            1L,
+        )
+        runGzipSdkSourceCompressionTest(
+            "Lorem ipsum dolor sit amet, dolorem corpora iracundia has ea, duo cu stet alterum scriptorem, " +
+                "et qui putent tractatos. Ne epicurei gloriatur pro, et ornatus consulatu necessitatibus qui. " +
+                "Veri eripuit feugiat sed no, dicat ridens id quo. Mei ne putent impedit antiopam. Ad libris " +
+                "assueverit his. Quo te brute vitae iuvaret, ut nibh bonorum sea. Mel altera vocibus ei, no vel" +
+                " tantas postea doming.",
+            1005L,
+        )
+    }
```

Comment:
correctness: these repeated `runXTest` should probably be parameterized, see other tests in the SDK which use `@ParameterizedTest` annotation

Reply:
That only works on JVM FYI